js/contact_table.js

```diff
@@ -162,6 +195,68 @@ function doDeleteContact(contactName, contactPhone, contactEmail, row) {
     }
 }
 
+//update contact
+function doUpdateContact(contactId, userId){
+    const UPDATECONTACT_ENDPOINT = API_URL + "/UpdateContact.php";
+
+    //cannot update contact if we cannot find userId
+    if (userId == null) {
+        console.log("Error in updating contact: cant find userId");
+        return;
+    }
+    //Grab the contactID
+    let xhr2 = XMLHttpRequest();
+    //gotta figure out how to get from api
+    xhr2.open("GET");
+    let contactId = xhr2.responseText;
```

Comment:
What do these lines do? Seems to me that parsing the response text before any response is sent will crash the code